src/test/java/net/minecraftforge/debug/gameplay/DynamicBannerEventTest.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016-2018.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.debug.gameplay;
+
+import java.awt.image.BufferedImage;
+import java.awt.image.ColorModel;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.imageio.ImageIO;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.BannerTextures;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.item.EnumDyeColor;
+import net.minecraft.item.ItemBanner;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.nbt.NBTTagList;
+import net.minecraft.tileentity.BannerPattern;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraftforge.client.event.BufferedImageLoadEvent;
+import net.minecraftforge.common.MinecraftForge;
+import net.minecraftforge.common.util.EnumHelper;
+import net.minecraftforge.fml.common.Mod;
+import net.minecraftforge.fml.common.event.FMLPreInitializationEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+@Mod(modid = DynamicBannerEventTest.MODID, name = "ForgeDebugDynamicBanner", version = DynamicBannerEventTest.VERSION, acceptableRemoteVersions = "*")
+public class DynamicBannerEventTest
+{
+    public static final String MODID = "forgedebugdynamicbanner";
+    public static final String VERSION = "1.0";
+
+    public static final CreativeTabs bannerTab = new CreativeTabBanners("dynbanner.banners");
+
+    @Mod.EventHandler
+    public void preInit(FMLPreInitializationEvent event)
+    {
+        MinecraftForge.EVENT_BUS.register(new EventHandler());
+        addBasicPattern("Y");
+    }
+
+    public static class EventHandler
+    {
+        @SideOnly(Side.CLIENT)
```

Comment:
This is generally bad practice, use `@EventBusSubscriber(value = Side.CLIENT, modid = MODID)` on the class instead with static handler methods.

Reply:
Which part is bad practice. Can you please clarify? The event system has grown for a few years since I last looked at it. :P

Reply:
Think I got it.